flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -605,6 +610,50 @@ boolean isSerializingTimestamps() {
 		return tc == TimeCharacteristic.EventTime | tc == TimeCharacteristic.IngestionTime;
 	}
 
+	private void broadcastCheckpointBarrier(
```

Comment:
Maybe those methods that were moved from `OperatorChain` shouldn't be now implemented in the `StreamTask` but encapsulated somehow? Now they are making `StreamTask` class even more complicated compared to how it was before.

Reply:
Yes, I also thought the current `StreamTask` is overstaffing at the moment also besides `OperatorChain`. Let me think encapsulate it in another way.

Reply:
I already updated the codes for addressing this issue.